bundles/org.openhab.binding.omnilink/src/main/java/org/openhab/binding/omnilink/internal/handler/OmnilinkBridgeHandler.java

```diff
@@ -492,7 +493,7 @@ private void getSystemStatus() throws IOException, OmniNotConnectedException, Om
                 .append(String.format("%02d", status.getHour())).append(":")
                 .append(String.format("%02d", status.getMinute())).append(":")
                 .append(String.format("%02d", status.getSecond())).toString();
-        updateState(CHANNEL_SYSTEM_DATE, new DateTimeType(dateString));
+        updateState(CHANNEL_SYSTEM_DATE, new StringType(dateString));
```

Comment:
Leaving DateTimeType/StringType discussion aside for a moment, I wonder if something like this would have been a better approach:
```
        updateState(CHANNEL_SYSTEM_DATE,
                new DateTimeType(ZonedDateTime.ofLocal(LocalDateTime.of(status.getYear(), status.getMonth(),
                        status.getDay(), status.getHour(), status.getMinute(), status.getSecond()),
                        ZoneId.systemDefault(), null)));
```

Also, I can see that `SystemStatus` returned by the controller contains `daylightSavings`, but unfortunately that seems useless without also knowing the time zone. Do you need time zone/DST information from the controller, or simply from the openHAB system, so you can assume that system time returned by the controller is local time in openHAB's timezone?

The example is for simplicity using system time zone. Using `TimeZoneProvider` it's possible to get the time zone configured in openHAB.

Reply:
Perhaps, but I actually didn't write that code in the first place. I migrated the existing code and got it up to spec for inclusion in the openHAB addons proper repo and never really changed it. At this point it doesn't make a to much sense to have it be a date time type because of the issues discussed above but that has yet to be resolved, so for the moment I plan to leave it.